🌐 Translation Codebase overview (#98)
reactjs/fr.reactjs.org: content/docs/codebase-overview.md

if (!didWarnAboutMath) {
  warning(
    2 + 2 === 4,
    'Les Mathématiques ne fonctionnent pas auourd\'hui.'
  );
  didWarnAboutMath = true;
}
```

Les avertissements ne sont activés que dans la phase de développement. En production, ils sont complètement occultés. Si vous avez besoin d'interdire l'exécution d'une partie de code, utilisez plutôt le module `invariant` :

```js
var invariant = require('invariant');

invariant(
  2 + 2 === 4,

>   'Vous ne passerez pas !'

Author: Moins impressionnante peut être mais ne les blâmons pas trop pour une fois que les studios ont pris des vrais acteurs de doublage pour un film post 2000 !

Le seul autre moteur de rendu officiellement pris en charge est [`react-art`](https://github.com/facebook/react/tree/master/packages/react-art). Auparavant, il se trouvait dans un [répertoire GitHub](https://github.com/reactjs/react-art) séparé, mais nous l'avons déplacé dans l'arborescence source principale pour le moment.

>**Remarque :**
>
>Techniquement, le [`react-native-renderer`](https://github.com/facebook/react/tree/master/packages/react-native-renderer) est une couche très mince qui apprend à React à interagir avec la mise en oeuvre de React Native.  Le véritable code spécifique à la plate-forme, gérant les vues natives ainsi que ses composants, réside dans le [répertoire React Native](https://github.com/facebook/react-native).

### Réconciliateurs {#reconcilers}

> Même des moteurs de rendu très différents comme React DOM et React Native doivent partager beaucoup de logique. En particulier, l'algorithme [réconciliation](/docs/reconciliation.html) doit être aussi similaire que possible afin que le rendu déclaratif, les composants personnalisés, l'état, les méthodes de cycle de vie et les références fonctionnent de manière cohérente sur toutes les plateformes.

Reviewer: ```suggestion
Même des moteurs de rendu très différents comme React DOM et React Native doivent partager beaucoup de logique. En particulier, l'algorithme de [réconciliation](/docs/reconciliation.html) doit être aussi similaire que possible afin que le rendu déclaratif, les composants personnalisés, l'état local, les méthodes de cycle de vie et les refs fonctionnent de manière cohérente sur toutes les plates-formes.
```

- Ref ≠ référence : concept spécifique à React vs. concept générique JS (comme props vs. properties)
- Orthographe correcte de plate-forme (d'ailleurs utilisée jusqu'ici)

> Pour résoudre ce problème, différents moteurs de rendu partagent du code entre eux. Nous appelons cette partie de React un «réconciliateur». Lorsqu'une mise à jour telle que `setState()` est planifiée, le réconciliateur appelle `render()` sur les composants de l'arborescence et les monte, les met à jour ou les supprime.

Reviewer: ```suggestion
Pour résoudre ce problème, différents moteurs de rendu partagent du code entre eux. Nous appelons cette partie de React un « réconciliateur ». Lorsqu'une mise à jour telle que `setState()` est planifiée, le réconciliateur appelle `render()` sur les composants de l'arborescence et les monte, les met à jour ou les démonte.
```

> Les réconciliateurs ne sont pas empaquetés séparément, car ils ne disposent actuellement d'aucune API publique. Au lieu de cela, ils sont exclusivement utilisés par les moteurs de rendu tels que React DOM et React Native.

Reviewer: ```suggestion
Les réconciliateurs ne sont pas empaquetés séparément, car ils ne disposent actuellement d'aucune API publique. Ils sont exclusivement utilisés par les moteurs de rendu tels que React DOM et React Native.
```

Ici le “instead” est superflu, autant le virer.

### Réconciliateur Stack {#stack-reconciler}

> Le réconciliateur «stack» est l'implémentation qui alimente React 15 et les versions antérieures. Nous avons depuis cessé de l'utiliser, tout ceci est décrit en détail dans la [prochaine section](/docs/implementation-notes.html).

Reviewer: ```suggestion
Le réconciliateur _“stack”_ est l'implémentation qui sous-tend React 15 et ses versions antérieures. Nous avons depuis cessé de l'utiliser, mais il est décrit en détail dans la [prochaine page](/docs/implementation-notes.html).
```

- On cite l'anglais avec les guillemets anglais, de préférence italisé
- « Section » sous-entend la page courante.

### Réconciliateur Fiber {#fiber-reconciler}

> Le réconciliateur «fiber» est un nouvel effort visant à résoudre les problèmes inhérents au réconciliateur de pile et à résoudre quelques problèmes de longue date. C’est le réconciliateur par défaut depuis React 16.

Reviewer: ```suggestion
Le réconciliateur _“fiber”_ est une nouvelle tentative de résoudre les problèmes inhérents au réconciliateur “stack” en plus de quelques problèmes anciens. C’est le réconciliateur par défaut depuis React 16.
```

Ses objectifs principaux sont :

> * Possibilité de diviser le travail arrêtable en morceaux.

Reviewer: ```suggestion
* Capacité à diviser un travail interruptible en segments.
```

“Ability to” ≠ “Possibility of”. On sous-entend plus qu'une possibilité, mais l'ensemble des conditions nécessaires pour quelque chose, et le plus souvent son existence effective.

> * Possibilité de hiérarchiser, de rebaser et de réutiliser des travaux en cours.

Reviewer: ```suggestion
* Capacité à hiérarchiser, déplacer et réutiliser des travaux en cours.
```

> * Possibilité de se répartir entre parents et enfants pour subvenir aux besoins du rendu dans React.

Reviewer: ```suggestion
* Capacité à jongler entre parents et enfants pour exécuter une mise en page avec React.
```

> * Possibilité de retourner plusieurs éléments de `render()`.

Reviewer: ```suggestion
* Capacité à renvoyer plusieurs éléments depuis `render()`.
```

On ne retourne pas, on renvoie, tu te rappelles ?

> * Meilleur support pour les limites d'erreur.

Reviewer: ```suggestion
* Meilleure prise en charge des périmètres d'erreur.
```

Trads normalisées

> Vous pouvez en savoir plus sur l’architecture des Fiber React [ici](https://github.com/acdlite/react-fiber-architecture) et [ici](https://medium.com/react-in-depth/inside-fiber-in-depth-overview-of-the-new-reconciliation-algorithm-in-react-e1c04700ef6e). Bien qu’elles soient livrées avec React 16, les fonctionnalités asynchrones ne sont pas encore activées par défaut.

Reviewer: ```suggestion
Vous pouvez en apprendre davantage sur l’architecture React Fiber [ici](https://github.com/acdlite/react-fiber-architecture) et [ici](https://medium.com/react-in-depth/inside-fiber-in-depth-overview-of-the-new-reconciliation-algorithm-in-react-e1c04700ef6e). Bien qu’elles soient livrées avec React 16, les fonctionnalités asynchrones ne sont pas encore activées par défaut.
```

Trad normalisée

Son code source est situé dans [`packages/react-reconciler`](https://github.com/facebook/react/tree/master/packages/react-reconciler).

### Système d'événements {#event-system}

> React implémente un système d'événements synthétiques agnostique des moteurs de rendu et fonctionne à la fois avec React DOM et React Native. Son code source se trouve dans [`packages/events`](https://github.com/facebook/react/tree/master/packages/events).

Reviewer: ```suggestion
React implémente un système d'événements synthétiques indépendant du moteur de rendu, qui fonctionne à la fois avec React DOM et React Native. Son code source se trouve dans [`packages/events`](https://github.com/facebook/react/tree/master/packages/events).
```

Voici une [vidéo qui plonge en profondeur dans ce code](https://www.youtube.com/watch?v=dRo_egw7tBc) (66 minutes).

> ### Et ensuite ? {#what-next}

Reviewer: ```suggestion
### Et maintenant ? {#what-next}
```

> Lisez la [section suivante](/docs/implementation-notes.html) pour en savoir plus sur la mise en œuvre du réconciliateur avant React 16. Nous n'avons pas encore documenté les composants internes du nouveau réconciliateur.

Reviewer: ```suggestion
Lisez la [prochaine page](/docs/implementation-notes.html) pour en apprendre davantage sur l’implémentation du réconciliateur avant React 16. Nous n'avons pas encore documenté les détails internes du nouveau réconciliateur.
```
